pkg/imgpkg/bundle/bundle_images_lock.go

```diff
@@ -130,10 +140,12 @@ func (o *Bundle) buildAllImagesLock(throttleReq *util.Throttle, processedImgs *p
 			bundles = append(bundles, nestedBundles...)
 
 			// Adds Image to the resulting ImagesLock
+			typedImgRef := NewContentImageRef(imgRef,
+				len(nestedBundles) > 0, // nestedBundles have Bundles when the image is a bundle
+			)
```

Comment:
nit: this reads quite funky on multilines, mind squashing into a single line?

(same below for `processedImageRefs.AddImagesRef(`)